analyzers/src/SonarAnalyzer.VisualBasic/Rules/MethodsShouldNotHaveIdenticalImplementations.cs

```diff
@@ -49,5 +51,7 @@ protected override IEnumerable<MethodBlockSyntax> GetMethodDeclarations(SyntaxNo
 
         protected override SyntaxToken GetMethodIdentifier(MethodBlockSyntax method) =>
             method.SubOrFunctionStatement.Identifier;
+
+        protected override bool IsExcludedFromBeingExamined(ISymbol nodeContainingSymbol) => nodeContainingSymbol.Kind != SymbolKind.NamedType;
```

Comment:
You can provide the default implementation in the base class and override it only where it is needed (C#). WDYT?